File: Nvisibl.DataLibrary/Models/Chatroom.cs
﻿using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Nvisibl.DataLibrary.Models
{
    public class Chatroom
    {
        public int Id { get; set; }

        [MaxLength(100)]
        [Column(TypeName = "varchar(100)")]
        public string Name { get; set; } = string.Empty;

        [Required]
        public bool IsShared { get; set; }

        public IList<ChatroomUser> Users { get; set; } = new List<ChatroomUser>();

        public IList<Message> Messages { get; set; } = new List<Message>();
    }
}

File: Nvisibl.WebUI/src/models/chatroom.ts
import User from './user';

export default class Chatroom {
    constructor(id: number, name: string, isShared: boolean, users: User[]) {
        this.id = id;
        this.name = name;
        this.isShared = isShared;
        this.users = users;
    }

    public readonly id: number;

    public readonly name: string;

    public readonly isShared: boolean;

    public readonly users: User[];
}
